### ISSUE TITLE
LSP missing tokens: pub struct entire block

### ISSUE THREAD

**Issue by sdankel** [FuelLabs/sway#3720]
Unable to access span information for `pub struct` entire block

![image](https://user-images.githubusercontent.com/47993817/211123800-a181d214-3d59-4729-94f8-78397b6b654b.png)

**Comment by JoshuaBatty** [comment]
Not 100% sure what happened here. When I run the above code I can see that we are in fact collecting all of the tokens.

<img width="205" alt="Screen Shot 2023-01-10 at 2 22 07 pm" src="https://user-images.githubusercontent.com/1289413/211454873-194fc889-f543-44f3-b314-6956551dda63.png">

I think we should be able to close this issue unless there is something i'm missing?

**Comment by JoshuaBatty** [comment]
I've worked out what happened here. This struct definition is happening in the following test.

`"test/src/e2e_vm_tests/test_programs/should_pass/language/dependencies"` and is located with `src/inner/bar.sw`. 

As this dependency is never imported into the project then this code never passes dead code analysis. As such, none of these tokens make it into the AST so we aren't able to collect them in the language server. 

This is the same way `rust-analyzer` works so I don't think any changes need to happen. Closing this issue as solved.
